app/gui2/shared/languageServer.ts

```diff
@@ -152,8 +153,13 @@ export class LanguageServer extends ObservableV2<Notifications & TransportEvents
     this.client.onError((error) => {
       console.error('Unexpected Language Server connection error:', error)
     })
-    transport.on('error', (error) => console.error('Language Server transport error:', error))
+    transport.on('error', (error) => {
+      if (this.shouldReconnect) {
+        console.error('Language Server transport error:', error)
+      }
```

Comment:
I think we should always print this? Error is an error

Reply:
possibly. i *think* i added this because it logged an error upon closing for some reason, but it is entirely possible that one or more of these guards are no longer needed

Reply:
perhaps instead the event handler should be unregistered using `.off()` like `reinitializeCb`? although i'm not sure it's semantically different than printing things only conditionally